File: sdk/android/src/main/java/com/dualler/sdk/view/ViewManager.kt
package com.dualler.sdk.view

import android.content.Context
import android.view.View
import android.widget.ImageView
import android.widget.LinearLayout
import android.widget.TextView
import android.util.TypedValue
import android.graphics.Color

/**
 * Manages native UI views that correspond to DSL nodes.
 */
class ViewManager(private val context: Context) {
    private val views = mutableMapOf<String, View>()

    fun createView(nodeType: String, props: Map<String, Any?>): View {
        return when (nodeType) {
            "view" -> createContainerView(props)
            "text" -> createTextView(props)
            "image" -> createImageView(props)
            else -> createDefaultView(props)
        }
    }

    fun registerView(id: String, view: View) {
        views[id] = view
    }

    fun getView(id: String): View? = views[id]

    fun removeView(id: String) {
        views.remove(id)
    }

    private fun createContainerView(props: Map<String, Any?>): View {
        return LinearLayout(context).apply {
            orientation = LinearLayout.VERTICAL
        }
    }

    private fun createTextView(props: Map<String, Any?>): View {
        return TextView(context).apply {
            val textSize = (props["fontSize"] as? Number)?.toInt() ?: 14
            setTextSize(TypedValue.COMPLEX_UNIT_SP, textSize.toFloat())
            val color = (props["color"] as? String) ?: "#000000"
            setTextColor(Color.parseColor(color))
        }
    }

    private fun createImageView(props: Map<String, Any?>): View {
        return ImageView(context)
    }

    private fun createDefaultView(props: Map<String, Any?>): View {
        return View(context)
    }
}

File: sdk/ios/Sources/DuallerSDK/ViewManager.swift
import UIKit

/// Manages native UI views mapped from DSL nodes
public final class ViewManager {
    private var views: [String: UIView] = [:]

    public func createView(nodeType: String, props: [String: Any?]) -> UIView {
        switch nodeType {
        case "view":
            return UIView()
        case "text":
            let label = UILabel()
            if let fontSize = props["fontSize"] as? CGFloat {
                label.font = label.font.withSize(fontSize)
            }
            if let colorStr = props["color"] as? String {
                label.textColor = UIColor(hex: colorStr)
            }
            return label
        case "image":
            return UIImageView()
        default:
            return UIView()
        }
    }

    public func registerView(id: String, view: UIView) {
        views[id] = view
    }

    public func getView(id: String) -> UIView? {
        views[id]
    }

    public func removeView(id: String) {
        views.removeValue(forKey: id)
    }
}
